src/components/SearchBar.js

```diff
@@ -0,0 +1,114 @@
+import styled from 'styled-components';
+import { AiOutlineSearch } from 'react-icons/ai';
+
+function SearchBar({ getRegion, getCountry }) {
+    const debounce = (func, timeout = 300) => {
+        let timerId;
+
+        return (...args) => {
+            if (timerId) {
+                clearTimeout(timerId);
+            }
+
+            timerId = setTimeout(() => {
+                func(...args);
+            }, timeout);
+        };
+    };
```

Comment:
I would suggest defining this outside of your component.

Because this is not part of what the component should do, a component should render data and add bindings.